src/components/name-input.js

```diff
@@ -10,18 +10,38 @@ import {
 } from '../styles/pages/welcome';
 import ArrowIcon from '../icons/white-arrow.png';
 
-export const NameInput = ({ nextPage }) => (
-  <NameInputView style={{ flex: 1, flexDirection: 'row' }}>
-    <TextInputWrapper>
-      <NameInputText placeholder="What&apos;s your name?" />
-    </TextInputWrapper>
-    <NextButton onPress={() => nextPage()}>
-      <ButtonText>Next</ButtonText>
-      <WhiteArrowIcon source={ArrowIcon} />
-    </NextButton>
-  </NameInputView>
-);
+export default class NameInput extends React.Component {
+  static propTypes = {
+    handleOnFocus: PropTypes.func.isRequired,
+    handleMarginTop: PropTypes.number.isRequired,
+    handleOnPress: PropTypes.func.isRequired,
+    isFocused: PropTypes.bool.isRequired,
+  };
 
-NameInput.propTypes = { nextPage: PropTypes.func.isRequired };
-
-export default NameInput;
+  render() {
+    const {
+      handleOnFocus,
```

Comment:
Use `on[Event]` to indicate the component is triggering an event; `handle[Event]` to handle it